src/daft-plan/src/optimization/rules/push_down_filter.rs

```diff
@@ -116,26 +115,55 @@ impl OptimizerRule for PushDownFilter {
                             return Ok(Transformed::No(plan));
                         }
                         let new_predicate = external_info.pushdowns().filters.as_ref().map(|f| predicate.and(f)).unwrap_or(predicate.clone());
-                        let partition_filter = if let ExternalInfo::Scan(ScanExternalInfo {scan_op,  ..}) = &external_info {
+                        let PredicateGroups { partition_filter, non_identity_partition_filter_for_data, partition_and_data_filter, data_only_filter } = if let ExternalInfo::Scan(ScanExternalInfo {scan_op,  ..}) = &external_info {
```

Comment:
given that we now bucket things into these categories and add a filter manually, we can now split the input expression, filtering out the ones that `has_udf`.